main/ui/src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java

```diff
@@ -73,6 +73,20 @@
 		this.successScene = successScene;
 		this.invalidMountPointScene = invalidMountPointScene;
 		this.errorComponent = errorComponent;
+
+		setOnScheduled(event -> vault.setState(VaultState.PROCESSING));
+		setOnSucceeded(event -> vault.setState(VaultState.UNLOCKED));
+		setOnCancelled(event -> vault.setState(VaultState.LOCKED));
```

Comment:
afaik there can only be a _single_ listener and this is a public method. I.e. if someone set a different "onSucceeded" listener, the vault state would no longer be updated correctly.

In other words: Overwriting the proteceted `succeeded()` (etc) methods was a deliberate choice to have the listener still available to the caller without interference.

Reply:
>afaik there can only be a single listener and this is a public method. I.e. if someone set a different "onSucceeded" listener, the vault state would no longer be updated correctly.

That's correct

I wanted to decide on one common way of doing this (overridden method or listener) and was under the belief that either would cause the same problems when being overridden. I didn't notice the `protected` visibility.

Reply:
Resolved by 0acc8ddb0a667930a98caacac4ecac8426a83c53.